tests/test_annotation_stores.py

```diff
@@ -185,7 +200,278 @@ def _fill_store(
+# Annotation Tests
+# ----------------------------------------------------------------------
+
+
+def test_annotation_init_both_shapely_wkb():
+    """Init annotation with both wkb and geometry."""
+    with pytest.raises(ValueError, match="geometry and wkb"):
+        _ = Annotation(geometry=Point(0, 0), wkb=Point(0, 0).wkb)
+
+
+def test_annotation_init_neither_shapely_wkb():
+    """Init annotation with neither wkb nor geometry."""
+    with pytest.raises(ValueError, match="geometry or wkb"):
+        _ = Annotation()
+
+
+def test_polygon_annotation_from_wkb():
+    """Test creating a polygon annotation from wkb."""
+    coords = [[0, 0], [1, 1], [2, 0]]
+    wkb = Polygon(coords).wkb
+    ann = Annotation(wkb=wkb)
+    assert ann.geometry == shapely.from_wkb(wkb)
+    assert ann.properties == {}
+    assert ann.geometry.type == str(GeometryType.POLYGON)
+
+
+def test_polygon_annotation_from_shapely():
+    """Test creating an annotation from shapely polygon then accessing coords."""
+    polygon = Polygon([[0, 0], [1, 1], [2, 0]])
+    ann = Annotation(geometry=polygon)
+    assert ann.geometry == polygon
+    assert ann.properties == {}
+    assert ann.geometry.type == str(GeometryType.POLYGON)
+
+
+def test_annotation_lazy_geometry():
+    """Test that the geometry is not created until it is accessed."""
+    ann = Annotation(wkb=Polygon([[0, 0], [1, 1], [2, 0]]).wkb)
+    assert ann._wkb is not None
+    assert ann._geometry is None
+    _ = ann.geometry
+    assert ann._geometry is not None
+
+
+def test_lazy_geometry_type():
+    """Test that getting geometry type from WKB does not create geometry."""
+    ann = Annotation(wkb=Polygon([[0, 0], [1, 1], [2, 0]]).wkb)
+    assert ann.geometry.type == str(GeometryType.POLYGON)
+    assert ann._geometry is not None
+
+
+def test_annotation_lazy_to_wkb():
+    """Test that the geometry is not created until it is accessed."""
+    polygon = Polygon([[0, 0], [1, 1], [2, 0]])
+    ann = Annotation(wkb=polygon.wkb)
+    _ = ann.to_wkb()
+    assert ann._geometry is None
+
+
+def test_annotation_lazy_wkb():
+    """Test that the geometry is not created by accessing WKB."""
+    polygon = Polygon([[0, 0], [1, 1], [2, 0]])
+    ann = Annotation(wkb=polygon.wkb)
+    _ = ann.wkb
+    assert ann._geometry is None
+
+
+def test_annotation_to_wkb_equals_wkb():
+    """Test that to_wkb() and wkb are equal."""
+    polygon = Polygon([[0, 0], [1, 1], [2, 0]])
+    ann = Annotation(wkb=polygon.wkb)
+    assert ann.to_wkb() == ann.wkb
+
+
+def test_annotation_point_wkb_coords():
+    """Test generating coordinates from WKB for Point.
+
+    This should produce coordinates without generating a Shapely
+    geometry.
+    """
+    point = Point([1, 2])
+    ann = Annotation(wkb=point.wkb)
+    coords = ann.coords
+    assert ann._geometry is None
+    assert isinstance(coords, np.ndarray)
+    assert len(coords) == 1
+    assert len(coords.shape) == 2
```

Comment:
Please can you specify a constant and then make a comparison? PLR2004

Reply:
I don't think that would be sensible for a test situation such as this. In normal code however yes.

Reply:
sure, we can add comments and then noqa in #677